src/codelabs/async-await/index.md

```diff
@@ -0,0 +1,498 @@
+provided functions. You don't need to implement `main()`.
+
+To simulate asynchronous operations, your code will call the following 
+functions, which are provided for you:
+
+|------------------+-----------------------------------+-------------|
+| Function         | Type signature                    | Description |
+|------------------|-----------------------------------|-------------|
+| getUsername()    | `Future<String> getUsername()`    |      Returns the name associated with the current user.|
+| logoutUser()    | `Future<String> logoutUser()`    |      Performs logout of current user and returns the username that was logged out.|
+{:.table .table-striped}
+
+Write the following:
+
+####  Part 1: `addHello()` 
+* Write a function `addHello()` that takes a single String argument 
+<!-- try to avoid should -->
+* `addHello()` returns its String argument surrounded by the text ‘Hello \<string\>'
+
+####  Part 2: `greetUser()` 
+* Write a function `greetUser()` that takes no arguments.
+* To obtain the username, `greetUser()` calls the provided asynchronous 
+function `getUsername()`
+* `greetUser()` creates a greeting for the user by calling `addHello()`, 
+passing it the username, and returning the result. 
+  * For example, if the username is "Jenny", `greetUser()` should create and
+   return the greeting "Hello Jenny"
+
+####  Part 3: `sayGoodbye()` 
+<!-- pull out descriptions of  -->
+* Write a function `sayGoodbye()` that does the following:
+* `sayGoodbye()` takes no arguments.
+* `sayGoodbye()` catches any errors.
+* `sayGoodbye()` calls the provided asynchronous function `logoutUser()`.
+<!-- italicize <result> -->
+* If `logoutUser()` succeeds, `sayGoodbye()` returns the string "\<result\> 
+Thanks, see you next time" where \<result\> is the String value returned by calling `logoutUser()`.
+
+    <!-- [//]: https://gist.github.com/f601d25bc2833c957186e3c6bf71effc -->
+<iframe 
+  src="https://dartpad.dev/experimental/embed-new-dart.html?id=f601d25bc2833c957186e3c6bf71effc&theme=dark"
+  frameborder="no"
+  height="550"
+  width="100%">
+</iframe>
+
+{{ site.alert.info }}
+  You might have noticed that the functions in the exercises don't have return types. That's because Dart can infer the return type for you. Omitting return types is fine when you're prototyping, but when you write production code, we recommend that you specify the return type.
+  Dart can [infer the return type](https://dart.dev/guides/language/sound-dart#type-inference) for you.
+{{ site.alert.end }}
+
+## What's next?
+
+* [Asynchronous programming: futures & async-await](/tutorials/language/futures) tutorial
+* [Dart's type system](/guides/language/sound-dart) (Includes more examples of type signatures with futures).
+* [Preferring signatures in function type annotations](/guides/language/effective-dart/design#prefer-signatures-in-function-type-annotations)
+* [Effective Dart](/guides/language/effective-dart)
```

Comment:
Does this tutorial https://dart.dev/tutorials/language/futures cover more content than this codelab? If it's mostly repeating, maybe it's "other resources' rather than "next step". 

--
In previous UX study, users said they want to learn more about how Dart handles threads, controlling flow/loop, scheduling, and more complex asynchronous tasks other than grabbing values. It seems like Dart uses `streams`, `Isolates`, `event loops`. Hence, we can consider adding these two docs as next step.

- [Asynchronous programming: streams](https://dart.dev/tutorials/language/streams) tutorial: show you how to work with sequence of asynchronous events.
- [Dart asynchronous programming: Isolates and event loops](https://medium.com/dartlang/dart-asynchronous-programming-isolates-and-event-loops-bffc3e296a6a) (link to an external doc on Dart's Medium)